synapse/handlers/room.py

```diff
@@ -576,6 +577,10 @@ async def create_room(
                 values to go in the body of the 'join' event (typically
                 `avatar_url` and/or `displayname`.
 
+            requested_room_id: Allow callees to request a particular room ID.
+                This is useful for testing event persistence sharding, and
+                *should not* be exposed to clients.
+
```

Comment:
this feels like a bit of a footgun waiting to be misused. Can you call `store_room` and `_send_events_for_new_room` from the test?

Reply:
alternatively: patch out `_generate_room_id`.